### PR TITLE
Add external resources for Python programming introductions

### PR DESCRIPTION
**Pull request by dominiquesydow**
## Description

Add external references to Python introduction material to TeachOpenCADD (README and website).

## Todos

- [x] Collect resources for Python introduction courses
  - AI in Medicine repository https://github.com/volkamerlab/ai_in_medicine
  - Python for Chemists repository https://github.com/GDChCICTeam/python-for-chemists
  - MolSSI resources  http://education.molssi.org/resources.html
- [x] Add Python programming intro resources to README
- [x] Add Python programming intro resources to website

## Questions

- Link to GDCh/CIC team?

## Status

- [ ] Ready to go

### CODE REVIEW

**Comment by dominiquesydow** [comment]
@AndreaVolkamer and @jaimergp, since we discuss this in the book chapter, I suggest to add a section to the README where we list resources for Python programming introductions. 

Could you please check if you like this? https://github.com/volkamerlab/teachopencadd/blob/292258a8695b92a4b21e2477343e16b03d9797b0/README.md#python-programming-introduction

Once this text is finalized, I suggest to also copy this section to the TeachOpenCADD website: https://projects.volkamerlab.org/teachopencadd/

File: README.md [hunk 1]
```diff
@@ -59,6 +59,16 @@ If you'd like to execute the provided notebooks, we offer two possibilities:
...
 
+## Python programming introduction
+
+The TeachOpenCADD platform is not a Python programming course from scratch but teaches how to solve tasks in cheminformatics and structural bioinformatics programmatically. 
+If you wish to get started with a low level Python programming introduction before diving into the TeachOpenCADD material, here are a few great resources to do so:
+
+- [Ai in Medicine course](https://github.com/volkamerlab/ai_in_medicine) by the [Volkamer Lab](https://volkamerlab.org/) and [Ritter Lab](https://psychiatrie-psychotherapie.charite.de/en/research/neuroimaging_and_neurotechnology/machine_learning/) at the Charité: A low-level introduction on Python basics, Jupyter Notebook, and important data science packages such as Pandas, Matplotlib and Scikit-learn.
+- [Python for Chemists course](https://github.com/GDChCICTeam/python-for-chemists) by the GDCh/CIC team: A crash-course introduction to Python for natural scientists.
```
Comment (jaimergp) [inline README.md:68]:
```suggestion
- [Python for Chemists course](https://github.com/GDChCICTeam/python-for-chemists) by the [GDCh/CIC](https://www.gdch.de/netzwerk-strukturen/fachstrukturen/computer-in-der-chemie-cic.html) team: A crash-course introduction to Python for natural scientists.
```

File: README.md [hunk 2]
```diff
@@ -59,6 +59,16 @@ If you'd like to execute the provided notebooks, we offer two possibilities:
...
+- [MolSSI Education Resources](http://education.molssi.org/resources.html) by [The Molecular Sciences Software Institute](https://molssi.org/): A collection of tutorials on basic Python programming and data analysis but also more advanced material on software development and computational molecular science.
+
```
Comment (jaimergp) [inline README.md:70]:
```suggestion
- [Core lessons](https://software-carpentry.org/lessons/) by the [Software Carpentry](https://software-carpentry.org/): introduction to Python, Git, command line interfaces and more.
```

File: README.md [hunk 3]
```diff
@@ -59,6 +59,16 @@ If you'd like to execute the provided notebooks, we offer two possibilities:
 
 If you prefer to work in the context of a graphical interface, talktorials T001-T008 are also available as [KNIME workflows](https://hub.knime.com/volkamerlab/space/TeachOpenCADD/TeachOpenCADD). Questions regarding this version should be addressed using the "Discussion section" available at [this post](https://forum.knime.com/t/teachopencadd-knime/17174). You might need to create a KNIME account.
 
+## Python programming introduction
+
+The TeachOpenCADD platform is not a Python programming course from scratch but teaches how to solve tasks in cheminformatics and structural bioinformatics programmatically. 
+If you wish to get started with a low level Python programming introduction before diving into the TeachOpenCADD material, here are a few great resources to do so:
+
+- [Ai in Medicine course](https://github.com/volkamerlab/ai_in_medicine) by the [Volkamer Lab](https://volkamerlab.org/) and [Ritter Lab](https://psychiatrie-psychotherapie.charite.de/en/research/neuroimaging_and_neurotechnology/machine_learning/) at the Charité: A low-level introduction on Python basics, Jupyter Notebook, and important data science packages such as Pandas, Matplotlib and Scikit-learn.
```
Comment (jaimergp) [inline README.md:67]: I am not sure about the use of "low-level" here. I think you mean "from scratch" and "with no prior experience", but I would understand "low-level" as the internal details of Python, how it works, binds to C, and so on...

**Comment by jaimergp** [comment]
Thanks for the write up, @DominiqueSydow! 🥳 I added a few comments to clarify the intent of the paragraph as well as two more resources that might be useful for the reader.

**Comment by AndreaVolkamer** [comment]
I like it a lot, thanks @dominiquesydow and @jaimergp! 
Two minor suggestions:
* can you add a link to GDCh/CIC: https://en.gdch.de/main-navi/network-structures/divisions/computers-in-chemistry.html
* and should we link (somewhere) to the RDKit UGM resources (it is not about python programming, but about resources in the field: https://github.com/rdkit/UGM_2020/blob/master/info/curated_list_of_resources.md

File: README.md (hunk 3 above)
Comment (AndreaVolkamer) [reply README.md:67]: Agree.

**Comment by jaimergp** [comment]
> can you add a link to GDCh/CIC

I had added this to my suggestions, but the German version. Updated to point to the English version.

**Comment by dominiquesydow** [comment]
Thanks a lot @AndreaVolkamer  and @jaimergp - please take one more look: https://github.com/volkamerlab/teachopencadd/blob/e3895800bd2c21ea1beb5cae8059a86444e05fdb/README.md

If ok, then I will add this also to the website.

**Comment by jaimergp** [comment]
Perfect! And yes, go ahead!

**Comment by AndreaVolkamer** [comment]
Super minor adaption: Ai in medicine -> AI in medicine (capital I)

**Comment by dominiquesydow** [comment]
@AndreaVolkamer, this is how the website will look like - what do you think?
image

**Comment by jaimergp** [comment]
Works for me! Watch the punctuation, though. In the first list, each item ends with periods, but the 2nd one does not.

**Comment by AndreaVolkamer** [comment]
The punctuation would work for me, because in the first list we have rather sentences, in the second list simple enumeration

**Comment by dominiquesydow** [comment]
Hi @jaimergp,

Do I need to take care of these fails?

Sphinx CI fails because of

```
Getting links from: http://localhost:8000/talktorials/T001_query_chembl.html
├─BROKEN─ http://doi.org/10.6019/CHEMBL.database.27 (BLC_UNKNOWN)
```

Link does exist.

---

Black CI fails because of

```
error: cannot format teachopencadd/talktorials/T002_compound_adme/talktorial.ipynb: INTERNAL ERROR: Black produced different code on the second pass of the formatter.
```

Any idea?

**Comment by jaimergp** [comment]
1. `http://doi.org/10.6019/CHEMBL.database.27` would need to be added [here](https://github.com/volkamerlab/teachopencadd/blob/ds-ai-in-med-ref/.github/workflows/docs.yml#L68).
2. I think this was fixed somewhere in the t11-revamp branch.

**Comment by dominiquesydow** [comment]
> I think this was fixed somewhere in the t11-revamp branch.

Not yet in `t11-revamp` branch but found it, thanks! https://github.com/volkamerlab/teachopencadd/issues/91

**Comment by dominiquesydow** [comment]
@jaimergp do I have your approval to merge this branch (I think the remaining CI fails are ok for now).

**Comment by jaimergp** [comment]
Go ahead!

**Pull request merged**